store/organizations.js

```diff
@@ -109,15 +187,52 @@ const $actions = {
 
     if (organizations) commit(UPDATE_CACHE, organizations.records);
 
-    return organizations;
+    return _.cloneDeep(organizations);
+  },
+
+  async getOrganization({ commit, getters }, { id, allowReadFromCache }) {
+    let organization = null;
+
+    if (allowReadFromCache && getters.isOrganizationCached(id)) {
+      organization = getters.getOrganization(id);
+    } else {
+      organization = await this.$kronosApi.getOrganization(id);
+      if (organization) commit(UPDATE_CACHE, [ organization ]);
+    }
+
+    return _.cloneDeep(organization);
+  },
+
+  async createOrganization({ commit }, organization) {
+    const result = await this.$kronosApi.createOrganization(organization);
+
+    if (result) commit(UPDATE_CACHE, [ result ]);
+
+    return _.cloneDeep(result);
   },
 
-  async getOrganizationById({ commit }, id) {
-    const organization = await this.$kronosApi.getOrganization(id);
+  async updateOrganization({ commit }, organization) {
+    const result = await this.$kronosApi.updateOrganization(organization);
+
+    if (result) commit(UPDATE_CACHE, [ result ]);
+
+    return _.cloneDeep(result);
+  },
 
-    if (organization) commit(UPDATE_CACHE, [ organization ]);
+  async deleteOrganization({ commit }, organizationId) {
+    await this.$kronosApi.deleteOrganization(organizationId);
 
-    return organization;
+    commit(REMOVE_CACHE, organizationId);
+  },
+
+  async getOrganizationContacts({ commit, getters }, { id, allowReadFromCache }) {
```

Comment:
Change method signature for the one specified int the trello card.
Using two parameters and default value. not an object
```suggestion
  async getOrganizationContacts({ commit, getters }, id, allowReadFromCache=false) {
```

Reply:
Hi @sbilodeau, 

Actually vuex action will only accept two parameters... One is context and second is payload... 

if we pass more parameters we will get undefined value of those parameters... 

So, I have passed it as a object.

Reply:
ok